inference-engine/thirdparty/movidius/XLink/shared/src/XLinkSemaphore.c

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "XLinkSemaphore.h"
+#include "XLinkErrorUtils.h"
+#include "XLinkLog.h"
+
+static pthread_mutex_t ref_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_cond_t ref_cond = PTHREAD_COND_INITIALIZER;
+
+int XLink_sem_init(XLink_sem_t* sem, int pshared, unsigned int value)
+{
```

Comment:
This is probably useful:

```suggestion
{
   XLINK_RET_ERR_IF(sem == null, -1);
```

Reply:
You can also check the `value`

` XLINK_RET_ERR_IF(value < -1, -1);`

Reply:
Thanks! Done

Reply:
Oops, `value` is unsigned :)